math/src/main/java/com/powsybl/math/timeseries/TimeSeries.java

```diff
@@ -9,29 +9,284 @@
 
     Iterator<P> iterator();
 
+    List<T> split(int n);
+
+    static <P extends AbstractPoint, T extends TimeSeries<P, T>> List<List<T>> split(List<T> timeSeriesList, int n) {
+        Objects.requireNonNull(timeSeriesList);
+        List<List<T>> splitList = new ArrayList<>(n);
+        for (int i = 0; i < n; i++) {
+            splitList.add(new ArrayList<>(timeSeriesList.size()));
+        }
+        for (T timeSeries : timeSeriesList) {
+            List<T> split = timeSeries.split(n);
+            for (int i = 0; i < n; i++) {
+                splitList.get(i).add(split.get(i));
+            }
+        }
+        return splitList;
+    }
+
+    static Map<Integer, List<TimeSeries>> parseCsv(Path file) {
+        return parseCsv(file, TimeSeriesConstants.DEFAULT_SEPARATOR);
+    }
+
+    static Map<Integer, List<TimeSeries>> parseCsv(String csv, char separator) {
+        try (BufferedReader reader = new BufferedReader(new StringReader(csv))) {
+            return parseCsv(reader, separator);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    static Map<Integer, List<TimeSeries>> parseCsv(Path file, char separator) {
+        try (BufferedReader reader = Files.newBufferedReader(file, StandardCharsets.UTF_8)) {
+            return parseCsv(reader, separator);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    static double parseDouble(String token) {
+        return token.isEmpty() ? Double.NaN : Double.parseDouble(token);
+    }
+
+    static String checkString(String token) {
+        return token.isEmpty() ? null : token;
+    }
+
+    class CsvParsingContext {
+        final List<String> names;
+
+        final TimeSeriesDataType[] dataTypes;
+        final Object[] values;
+
+        final List<ZonedDateTime> times = new ArrayList<>();
+
+        TimeSeriesIndex refIndex;
+
+        CsvParsingContext(List<String> names) {
+            this.names = names;
+            dataTypes = new TimeSeriesDataType[names.size()];
+            values = new Object[names.size()];
+        }
+
+        private static AssertionError assertDataType(TimeSeriesDataType dataType) {
+            return new AssertionError("Unexpected data type " + dataType);
+        }
+
+        private TDoubleArrayList createDoubleValues() {
+            TDoubleArrayList doubleValues = new TDoubleArrayList();
+            if (!times.isEmpty()) {
+                for (int j = 0; j < times.size(); j++) {
+                    doubleValues.add(Double.NaN);
+                }
+            }
+            return doubleValues;
+        }
+
+        private List<String> createStringValues() {
+            List<String> stringValues = new ArrayList<>();
+            if (!times.isEmpty()) {
+                for (int j = 0; j < times.size(); j++) {
+                    stringValues.add(null);
+                }
+            }
+            return stringValues;
```

Comment:
return new ArrayList<>(Collections.nCopies(times.size(), null));

Reply:
But it would add an extra copy of the list.